tests/integration/security/sds_tls_origination/util/util.go

```diff
@@ -152,31 +172,41 @@ func createSecret(credentialType string, cn, ns string, ic TLSCredential, isNotG
 // SetupEcho creates two namespaces client and server. It also brings up two echo instances server and
 // client in respective namespaces. HTTP and HTTPS port on the server echo are set up. Egress Gateway is set up in the
 // service namespace to handle egress for "external" calls.
-func SetupEcho(t test.Failer, ctx resource.Context) (echo.Instance, echo.Instance, namespace.Instance, namespace.Instance) {
-	clientNamespace := namespace.NewOrFail(t, ctx, namespace.Config{
+func SetupEcho(t test.Failer, ctx resource.Context, apps *EchoDeployments) {
+	apps.ClientNamespace = namespace.NewOrFail(t, ctx, namespace.Config{
 		Prefix: "client",
 		Inject: true,
 	})
-	serverNamespace := namespace.NewOrFail(t, ctx, namespace.Config{
+	apps.ServerNamespace = namespace.NewOrFail(t, ctx, namespace.Config{
 		Prefix: "server",
 		Inject: true,
 	})
 	cluster := ctx.Clusters().Default()
-	var internalClient, externalServer echo.Instance
-	echoboot.NewBuilder(ctx).
+	var internalClient, internalVM, externalServer echo.Instance
+	echos := echoboot.NewBuilder(ctx).
 		With(&internalClient, echo.Config{
 			Cluster:   cluster,
-			Service:   "client",
-			Namespace: clientNamespace,
+			Service:   "a",
```

Comment:
why only change the client to "a" but not the server?

Reply:
I still use single server workload in the test, and extend clients to cover vm workload. 
I would prefer to set up server on a VM in a follow up PR.